packages/jest-util/src/fake_timers.js

```diff
@@ -229,7 +229,26 @@ export default class FakeTimers {
         'Ran ' +
           this._maxLoops +
           ' timers, and there are still more! ' +
-          "Assuming we've hit an infinite recursion and bailing out...",
+          "Assuming we've hit an infinite recursion and bailing out." +
+          'The following timers are still pending:' +
+          Object.keys(timers)
+            .sort((left, right) => timers[left].expiry - timers[right].expiry)
+            .map(function(key, index) {
+              '\n\nsetTimeout - pending timer time ' +
```

Comment:
missing `return` here?

Reply:
i don't think a return is missing since `Object,keys(object).map` should work like an iterate.

Reply:
if you just want to just iterate, you should use `.forEach`.

But if you don't return here, the string is never shown to the user as it's lost inside the `map`. So you want `map` and the `return`.

Something like this (which still fails, but it's a start):

```diff
diff --git i/packages/jest-util/src/fake_timers.js w/packages/jest-util/src/fake_timers.js
index a1870352..1e0b09e7 100644
--- i/packages/jest-util/src/fake_timers.js
+++ w/packages/jest-util/src/fake_timers.js
@@ -234,16 +234,16 @@ export default class FakeTimers {
           Object.keys(timers)
             .sort((left, right) => timers[left].expiry - timers[right].expiry)
             .map(function(key, index) {
-              '\n\nsetTimeout - pending timer time ' +
+              return '\n\nsetTimeout - pending timer time ' +
                 timers[key].expiry -
-                this._now;
+                this._now +
               '\n' +
                 formatStackTrace(new Error().stack, this._global.setTimeout, {
                   noStackTrace: true,
                 }) +
                 '\n\nsetImmediate - pending timer time ' +
                 timers[key].expiry -
-                this._now;
+                this._now +
               '\n' +
                 formatStackTrace(new Error().stack, this._global.setImmediate, {
                   noStackTrace: true,
```

Reply:
ah yeah, does make sense. i thought using object.keys would be nicer to iterate through an object since i also can use sort to get the pending timers that are left.